### ISSUE TITLE
restoring from snapshot is not cross-platform compatible

### ISSUE THREAD

**Issue by achubaty** [PredictiveEcology/Require#41]
After creating a snapshot file on Windows, trying to restore on Ubuntu:

```r
Require(packageVersionFile = "packages/pkgSnapshot.txt") ## FAILS
```

Multiple failures:
- snapshot file uses absolute paths with machine-specific package location and is version-specific (e.g., `"D:/GitHub/LandR-Manual/packages/x86_64-w64-mingw32/4.0"` is not valid on Ubuntu; should use relative paths `packages` and not specify the compiler platform and probably not the version)
- restoring snapshot installs binaries of pkgs needed from source (note I am using RSPM for binary packages by default);
- some packages are archived on CRAN (e.g., `RandomFields` and `gdalUtils`) but are dependencies of several required packages

Reprex: try restoring from this snaphot

```r
library(Require)
pkgPath <- file.path(tempdir(), "packages")
checkPath(pkgPath, create = TRUE)
download.file("https://raw.githubusercontent.com/PredictiveEcology/LandR-Manual/30a51761e0f0ce27698185985dc0fa763640d4ae/packages/pkgSnapshot.txt", destfile = file.path(pkgPath, "pkgSnapshot.txt"))
Require(packageVersionFile = file.path(pkgPath, "pkgSnapshot.txt"))
```

**Comment by achubaty** [comment]
@eliotmcintire has the absolute path issue been addressed in package snapshots?

**Comment by eliotmcintire** [comment]
Super sneaky... that package snapshot includes a version of `terra` (`terra (==1.6-16)`) that is not hosted on the [CRAN Archive](https://cran.r-project.org/src/contrib/Archive/terra/), *and* [MRAN was down for 2 weeks](https://mran.revolutionanalytics.com/snapshot/) when that version might have been there. This reprex is **particularly difficult** as there is no standard record that `terra` ever had that version.

**Comment by eliotmcintire** [comment]
More sneaky bits -- there are several packages that are in that package snapshot that do not have their dependencies (Imports/Depends) as part of the snapshot. So, in other words, the particular snapshot is technically impossible to exist. Clearly there is a path dependency in *how* those packages were installed.

**Comment by achubaty** [comment]
has this been resolved in v0.1.4?

**Comment by eliotmcintire** [comment]
This example works in tests now.
